core/src/main/scala/org/apache/spark/deploy/rest/StandaloneRestServer.scala

```diff
@@ -139,7 +139,9 @@ private[rest] class StandaloneSubmitRequestServlet(
     val driverExtraLibraryPath = sparkProperties.get("spark.driver.extraLibraryPath")
     val superviseDriver = sparkProperties.get("spark.driver.supervise")
     val appArgs = request.appArgs
-    val environmentVariables = request.environmentVariables
+    // Filter SPARK_LOCAL environment variables from being set on the remote system.
+    val environmentVariables =
+      request.environmentVariables.filterNot(_._1.startsWith("SPARK_LOCAL"))
```

Comment:
I think this should also affect `SPARK_LOCAL_DIRS`, is that expected?

Reply:
Alright, I will check how it is used across the project. 
Just noted, In `LocalDirsSuite`, comments in `test("SPARK_LOCAL_DIRS override also affects driver") ` seems to corroborate with my intentions here.